Improper `DataFrame.__getitem__` call when using `DataFrame.merge`

## Describe the bug
When attempting to merge two dataframes via `arkodua.dataframe.DataFrame.merge`, arkouda will attempt to access a column in a dataframe with the value passed to the `how` parameter

## To Reproduce
### Reproducer:
```py
import arkouda as ak
import pandas as pd
import pytest


@pytest.fixture(scope="module", autouse=True)
def _init_arkouda():
    # TODO: Add your url here
    ak.connect(connect_url=...)
    ak.get_config()
    yield
    ak.disconnect()


@pytest.mark.parametrize("df_init, merge", [
    pytest.param(
        pd.DataFrame,
        lambda df_1, df_2: df_1.merge(df_2, on=["idx"], how="inner"),
        id="Pandas via `DataFrame.merge`"),
    pytest.param(
        pd.DataFrame,
        lambda df_1, df_2: pd.merge(df_1, df_2, on=["idx"], how="inner"),
        id="Pandas via `pandas.merge`"),
    pytest.param(
        ak.DataFrame,
        lambda df_1, df_2: df_1.merge(df_2, on=["idx"], how="inner"),
        id="Arkouda via `DataFrame.merge`"),
    pytest.param(
        ak.DataFrame,
        lambda df_1, df_2: ak.merge(df_1, df_2, on=["idx"], how="inner"),
        id="Arkouda via `arkouda.merge`"),
])
def test_merge(df_init, merge):
    df_first = df_init({
        "idx": list(range(10)),
        "val": list(range(10)), })
    df_second = df_init({
        "idx": list(range(10)),
        "val": list(map(lambda x: x + 3, range(10))), })
    df_merged = merge(df_first, df_second)
    assert (df_merged["val_y"] - df_merged["val_x"] == 3).all()
```

### Output
```py
====================================== test session starts ======================================
platform linux -- Python 3.13.0, pytest-8.3.5, pluggy-1.5.0 -- /tmp/drozt/venv/bin/python3
cachedir: .pytest_cache
rootdir: /tmp/drozt
plugins: typeguard-2.10.0
collected 4 items

test_merge.py::test_merge[Pandas via `DataFrame.merge`] PASSED                            [ 25%]
test_merge.py::test_merge[Pandas via `pandas.merge`] PASSED                               [ 50%]
test_merge.py::test_merge[Arkouda via `DataFrame.merge`] FAILED                           [ 75%]
test_merge.py::test_merge[Arkouda via `arkouda.merge`] PASSED                             [100%]

==================================== short test summary info ====================================
FAILED test_merge.py::test_merge[Arkouda via `DataFrame.merge`] - KeyError: "Invalid column name
'inner'."
============================ 1 failed, 3 passed, 1 warning in 3.58s =============================
```

### Traceback:
```py
=========================================== FAILURES ============================================
___________________________ test_merge[Arkouda via `DataFrame.merge`] ___________________________

df_init = <class 'arkouda.dataframe.DataFrame'>, merge = <function <lambda> at 0x7f354a21ec00>

    @pytest.mark.parametrize("df_init, merge", [
        pytest.param(
            pd.DataFrame,
            lambda df_1, df_2: df_1.merge(df_2, on=["idx"], how="inner"),
            id="Pandas via `DataFrame.merge`"),
        pytest.param(
            pd.DataFrame,
            lambda df_1, df_2: pd.merge(df_1, df_2, on=["idx"], how="inner"),
            id="Pandas via `pandas.merge`"),
        pytest.param(
            ak.DataFrame,
            lambda df_1, df_2: df_1.merge(df_2, on=["idx"], how="inner"),
            id="Arkouda via `DataFrame.merge`"),
        pytest.param(
            ak.DataFrame,
            lambda df_1, df_2: ak.merge(df_1, df_2, on=["idx"], how="inner"),
            id="Arkouda via `arkouda.merge`"),
    ])
    def test_merge(df_init, merge):
        df_first = df_init({
            "idx": list(range(10)),
            "val": list(range(10)), })
        df_second = df_init({
            "idx": list(range(10)),
            "val": list(map(lambda x: x + 3, range(10))), })
>       df_merged = merge(df_first, df_second)

test_merge.py:38:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
test_merge.py:24: in <lambda>
    lambda df_1, df_2: df_1.merge(df_2, on=["idx"], how="inner"),
venv/lib/python3.13/site-packages/typeguard/__init__.py:891: in wrapper
    retval = func(*args, **kwargs)
venv/lib/python3.13/site-packages/arkouda/dataframe.py:4645: in merge
    return merge(
venv/lib/python3.13/site-packages/typeguard/__init__.py:891: in wrapper
    retval = func(*args, **kwargs)
venv/lib/python3.13/site-packages/arkouda/dataframe.py:6247: in merge
    if not all(
venv/lib/python3.13/site-packages/arkouda/dataframe.py:6248: in <genexpr>
    isinstance(left[left_col], (pdarray, Strings))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self =    idx  val
0    0    0
1    1    1
2    2    2
3    3    3
4    4    4
5    5    5
6    6    6
7    7    7
8    8    8
9    9    9 (10 rows x 2 columns)
key = 'inner'

    def __getitem__(self, key):
        from arkouda.pandas.series import Series

        # convert series to underlying values
        # Should check for index alignment
        if isinstance(key, Series):
            key = key.values

        # Select rows using an integer pdarray
        if isinstance(key, pdarray):
            if key.dtype == akbool:
                key = arange(key.size)[key]
            result = {}
            for k in self._columns:
                result[k] = UserDict.__getitem__(self, k)[key]
            # To stay consistent with numpy, provide the old index values
            return DataFrame(initialdata=result, index=self.index.index[key])

        # Select rows or columns using a list
        if isinstance(key, (list, tuple)):
            result = DataFrame()
            if len(key) <= 0:
                return result
            if len({type(x) for x in key}) > 1:
                raise TypeError("Invalid selector: too many types in list.")
            if isinstance(key[0], str):
                for k in key:
                    result[k] = self[k]
                result._empty = False
                result._set_index(self.index)  # column lens remain the same. Copy the indexing
                return result
            else:
                raise TypeError(
                    "DataFrames only support lists for column indexing. "
                    "All list entries must be of type str."
                )

        # Select a single row using an integer
        if isinstance(key, int):
            result = {}
            row = array([key])
            for k in self._columns:
                result[k] = (UserDict.__getitem__(self, k)[row])[0]
            return Row(result)

        # Select a single column using a string
        elif isinstance(key, str):
            if key not in self.keys():
>               raise KeyError(f"Invalid column name '{key}'.")
E               KeyError: "Invalid column name 'inner'."

venv/lib/python3.13/site-packages/arkouda/dataframe.py:1110: KeyError
```


## Expected behavior
No `DataFrame.__getitem__` call with the `how` value as the `key` parameter should be made. The reproducer should pass like the pandas version of the test.

## Error Message
See the [provided traceback](#traceback) from the reproducer

## Is this a Blocking Issue
No. Calling `arkouda.merge` and manually passing in the first dataframe works as expected per the test output.

## `ak.get_config()` Output
```
{'ARROW_VERSION': '19.0.0',
 'HDF5Version': '1.14.4',
 'LocaleConfigs': [{'id': 0,
                    'maxTaskPar': 64,
                    'name': 'xxx',
                    'numPUs': 64,
                    'physicalMemory': 485501849600},
                   {'id': 1,
                    'maxTaskPar': 64,
                    'name': 'yyy',
                    'numPUs': 64,
                    'physicalMemory': 485501890560},
                   {'id': 2,
                    'maxTaskPar': 64,
                    'name': 'zzz',
                    'numPUs': 64,
                    'physicalMemory': 485501886464}],
 'ServerPort': 5555,
 'ZMQVersion': '4.3.5',
 'arkoudaVersion': 'v2025.01.13+104.ge936c6c69',
 'authenticate': False,
 'autoShutdown': False,
 'byteorder': 'little',
 'chplVersion': '2.5.0 (379d5c3266)',
 'distributionType': 'BlockDom(1,int(64),one,unmanaged DefaultDist)',
 'logChannel': 'CONSOLE',
 'logLevel': 'INFO',
 'maxArrayDims': 1,
 'maxTaskPar': 64,
 'numLocales': 3,
 'numPUs': 64,
 'physicalMemory': 485501849600,
 'pythonVersion': '3.13',
 'regexMaxCaptures': 20,
 'serverHostname': 'xxx',
 'serverInfoNoSplash': False}
```

## Additional context
Both `arkouda_server` and the python package were built off of commit e936c6c6924eb3ebaf9c7ed350b286a3cc1fe3e7

--- a/arkouda/dataframe.py
+++ b/arkouda/dataframe.py
@@ -4643,7 +4643,14 @@ class DataFrame(UserDict):
 
         """
         return merge(
-            self, right, on, how, left_suffix, right_suffix, convert_ints=convert_ints, sort=sort
+            self,
+            right,
+            on,
+            how=how,
+            left_suffix=left_suffix,
+            right_suffix=right_suffix,
+            convert_ints=convert_ints,
+            sort=sort,
         )
 
     @typechecked

--- a/tests/dataframe_test.py
+++ b/tests/dataframe_test.py
@@ -1254,6 +1254,37 @@ class TestDataFrame:
                 else:
                     assert (np.sort(from_ak) == np.sort(from_pd.astype(str))).all()
 
+    @pytest.mark.parametrize(
+        "df_init, merge",
+        [
+            pytest.param(
+                ak.DataFrame,
+                lambda df_1, df_2: df_1.merge(df_2, on=["idx"], how="inner"),
+                id="Arkouda via `DataFrame.merge`",
+            ),
+            pytest.param(
+                ak.DataFrame,
+                lambda df_1, df_2: ak.merge(df_1, df_2, on=["idx"], how="inner"),
+                id="Arkouda via `arkouda.merge`",
+            ),
+        ],
+    )
+    def test_merge_instance_and_module(self, df_init, merge):
+        df_first = df_init(
+            {
+                "idx": list(range(10)),
+                "val": list(range(10)),
+            }
+        )
+        df_second = df_init(
+            {
+                "idx": list(range(10)),
+                "val": list(map(lambda x: x + 3, range(10))),
+            }
+        )
+        df_merged = merge(df_first, df_second)
+        assert (df_merged["val_y"] - df_merged["val_x"] == 3).all()
+
     def test_isna_notna(self):
         df = ak.DataFrame(
             {

